configuration/secrets.rst

```diff
@@ -0,0 +1,319 @@
+them.
+
+.. _secrets-configuration:
+
+Configuration
+-------------
+
+In order to use secrets you have to enable the feature in the framework's
+
+configuration:
+.. configuration-block::
+
+    .. code-block:: yaml
+
+        # config/packages/framework.yaml
+        framework:
+            secrets:
+                # encrypted_secrets_dir: '%kernel.project_dir%/config/secrets/%kernel.environment%/'
+                # encryption_key: '%kernel.project_dir%/config/secrets/encryption_%kernel.environment%.key'
+
+    .. code-block:: xml
+
+            <!-- config/packages/framework.xml -->
+            <?xml version="1.0" encoding="UTF-8" ?>
+            <container xmlns="http://symfony.com/schema/dic/services"
+                xmlns:framework="http://symfony.com/schema/dic/framework"
+                xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+                xsi:schemaLocation="http://symfony.com/schema/dic/services https://symfony.com/schema/dic/services/services-1.0.xsd
+                    http://symfony.com/schema/dic/framework https://symfony.com/schema/dic/framework/framework-1.0.xsd"
+            >
+                <framework:config secret="%env(APP_SECRET)%">
+                    <framework:secrets/>
+                </framework:config>
+            </container>
+
+    .. code-block:: php
+
+        // config/packages/framework.php
+        $container->loadFromExtension('framework', [
+            'secrets' => true,
+        ]);
+
+.. tip::
+
+    the ``encryption_key`` configuration parameter accept both a path to an
+    encription key or the content of the itself. It allows you, for instance,
+    to store the content of the key in an environement variable and refer it
+    with ``%env(PRIVATE_KEY)%``.
+
+.. _secrets-generate-key:
+
+Generate an Encryption Key
+--------------------------
+
+Before creating a new ``secret``, you need to create ann ``encryption key``.
+This can be done with the provided commande ``secrets:generate-key``.
+
+.. code-block:: terminal
+
+    $ APP_ENV=prod php bin/console secrets:generate-key
+
+This command will generate a new ``encryption key`` in.
+``%kernel.project_dir%/config/secrets/encryption_%kernel.environment%.key``
+
+.. note::
+
+    In order to use Symfony's built-in Secret storage, you will need the
+    `libsodium`_ PHP extension or use the `sodium_compat`_ package.
+
+Symfony generates a key with a symetrics algorithm, meaning that this key could
+be used to both encrypt **and** decrypt secrets. The number of people who
+possess this key should be as small as possible.
+
+.. caution::
+
+    This file is sensitive and **must not** be commited nor publicly shared. Every
+    developpers and CI don't need that key. If the encryption key have been
+    exposed (ex-employee leaving for instance)you should consider regenerating a
+    new one.
+
+.. _secrets-add:
+
+Create a Secret
+---------------
+
+Once the ``encryption key`` generated, you can add new secret with the command
```

Comment:
encryption_key?

Reply:
hmm, `encryption key` is not configuration key nor a variable name. It's an important concept thaat I wanted to highlight (like the word `secret` is also quoted in this file)

Reply:
Understood. Then, it should be in bold (**encryption key**) and not as code (`encryption key`). The same goes for secret.